datafusion/core/src/logical_plan/builder.rs

```diff
@@ -432,8 +432,50 @@ impl LogicalPlanBuilder {
     }
 
     /// Apply a union, preserving duplicate rows
-    pub fn union(&self, plan: LogicalPlan) -> Result<Self> {
-        Ok(Self::from(union_with_alias(self.plan.clone(), plan, None)?))
+    pub fn union(&self, right_plan: LogicalPlan) -> Result<Self> {
+        let left_plan = self.plan.clone();
```

Comment:
This logic is moved from the `union_with_alias` function but no longer references the alias